awscli/customizations/s3/subcommands.py

```diff
@@ -481,6 +481,16 @@
     )
 }
 
+CHECKSUM_MODE = {
+        'name': 'checksum-mode', 'choices': ['ENABLED'],
+        'help_text': 'To retrieve the checksum, this mode must be enabled.'
+}
+
+CHECKSUM_ALGORITHM = {
+        'name': 'checksum-algorithm', 'choices': ['CRC32', 'SHA256', 'SHA1', 'CRC32C'],
```

Comment:
non-blocking: We should consider retrieving the choices from the service model instead of hardcoding the values here.

Reply:
Thanks for the feedback; could be worth looking into in a future PR.